charts/cluster-proportional-autoscaler/templates/configmap.yaml

```diff
@@ -12,5 +12,7 @@ metadata:
   name: {{ include "cluster-proportional-autoscaler.fullname" . }}
   namespace: {{ default .Release.Namespace .Values.options.namespace }}
 data:
+  {{- if $key }}
```

Comment:
Is a ConfigMap without any actual data in it considered a legit configuration? Shouldn't helm fail in that case as well?

/cc @krmichel

Reply:
What lint command is being run and what is the context for running it?

Reply:
Hi @krmichel, sorry for the late response. It's the default [helm lint ](https://helm.sh/docs/helm/helm_lint/).

> This command takes a path to a chart and runs a series of tests to verify that the chart is well-formed.

 I'm using it to check if the helm chart has any formatting errors as a CI test to ensure we are not pushing any invalid helm templates. To be more specific, I'm making a wrapper helm chart over this one to add additional templates for my use cases. As a workaround, I am adding a default config value so the lining passes but I would like to keep it empty so the user of the chart is required to think about his config and not just run the default.

Reply:
So, the "problem" is that `helm lint` doesn't observe the fail function before trying to render the template.  This change will allow lint to get far enough to print the fail message and exit with a 0 status.  In practice, anyone trying to actually use the chart won't be able to without specifying a config anyway so this change is fine.